pipeline/stage05_wave_characterization/scripts/local_macro_observables.py

```diff
@@ -0,0 +1,103 @@
+import neo
+import numpy as np
+import quantities as pq
+import matplotlib.pyplot as plt
+import os
+import argparse
+import scipy
+import pandas as pd
+from utils import load_neo, none_or_str, save_plot, AnalogSignal2ImageSequence
+import math 
+
+
+def calc_point_velocities(evts, DIM_X, DIM_Y):
+    N_Waves = len(np.unique(evts.labels)) 
+    WaveColl = np.ones([N_Waves, DIM_X, DIM_Y])*np.nan
+    WaveColl[np.int32(np.float64(evts.labels)), np.int32(evts.array_annotations['x_coords']), np.int32(evts.array_annotations['y_coords'])] = evts.times
+    WaveColl_2d = np.reshape(WaveColl, [np.shape(WaveColl)[0], np.shape(WaveColl)[1]*np.shape(WaveColl)[2]])
+   
+    #compute velocity
+    Tx = np.diff(WaveColl, axis = 1)[:, 0:np.shape(WaveColl)[1]-1, 0:np.shape(WaveColl)[2]-1]
+    Ty = np.diff(WaveColl, axis = 2)[:, 0:np.shape(WaveColl)[1]-1, 0:np.shape(WaveColl)[2]-1]
```

Comment:
Instead of only comparing to the next neighbors to one side, it is more robust (e.g. to nan values) to incorporate a wider range of neighbors, for example by using a corresponding kernel.